src/itemLogin/ItemLoginAuthorization.tsx

```diff
@@ -19,99 +14,58 @@ import ItemLoginScreen, { SignInPropertiesType } from './ItemLoginScreen.js';
-  useItem: (itemId?: string) => UseQueryResult<DiscriminatedItem>;
-  useItemLoginSchemaType: (args: {
-    itemId?: string;
-  }) => UseQueryResult<ItemLoginSchemaType>;
-  Error?: ReactElement;
-  memberIdInputId?: string;
+  currentAccount?: CurrentAccount | null;
+  item?: DiscriminatedItem;
+  itemLoginSchemaType?: ItemLoginSchemaType;
   usernameInputId?: string;
   signInButtonId?: string;
   passwordInputId?: string;
+  children?: ReactNode;
   ForbiddenContent?: ReactElement;
+  isLoading?: boolean;
 };
 
-const ItemLoginAuthorization =
-  ({
-    useCurrentMember,
-    useItem,
-    useItemLoginSchemaType,
-    itemId,
-    signIn,
-    Error: ErrorComponent,
-    usernameInputId,
-    signInButtonId,
-    passwordInputId,
-    ForbiddenContent = <ForbiddenText />,
-  }: ItemLoginAuthorizationProps) =>
-  (ChildComponent: () => JSX.Element) => {
-    const ComposedComponent = (): ReactElement => {
-      const {
-        data: user,
-        isLoading: isMemberLoading,
-        isError: isCurrentMemberError,
-      } = useCurrentMember();
-      const { data: itemLoginSchemaType } = useItemLoginSchemaType({ itemId });
-      const {
-        data: item,
-        isLoading: isItemLoading,
-        error: itemError,
-        isError: isItemError,
-      } = useItem(itemId);
-
-      if (isMemberLoading || (isItemLoading && !item)) {
-        // get item login if the user is not authenticated and the item is empty
-        return <CustomInitialLoader />;
-      }
-
-      // member should never trigger an error
-      // but can be empty
-      if (isCurrentMemberError) {
-        return (
-          ErrorComponent ?? <Alert severity='error'>An error occurred.</Alert>
-        );
-      }
+const ItemLoginAuthorization = ({
+  currentAccount,
+  item,
+  itemLoginSchemaType,
+  itemId,
+  signIn,
+  isLoading,
+  usernameInputId,
+  signInButtonId,
+  passwordInputId,
+  ForbiddenContent = <ForbiddenText />,
+  children,
+}: ItemLoginAuthorizationProps): ReactNode => {
+  if (isLoading) {
+    return <CustomInitialLoader />;
+  }
 
-      if (
-        isItemError &&
-        [
-          getReasonPhrase(StatusCodes.BAD_REQUEST),
-          getReasonPhrase(StatusCodes.NOT_FOUND),
-        ].includes((itemError as Error).message)
-      ) {
-        return (
-          ErrorComponent ?? <Alert severity='error'>An error occurred.</Alert>
-        );
-      }
+  // the item could be fetched without errors
+  // because the user is signed in and has access
+  // or because the item is public
+  if (item && item.id) {
+    return children;
+  }
 
-      // the item could be fetched without errors
-      // because the user is signed in and has access
-      // or because the item is public
-      if (item && item.id) {
-        return <ChildComponent />;
-      }
+  // signed out but can sign in with item login
+  if ((!currentAccount || !currentAccount.id) && itemLoginSchemaType) {
```

Comment:
This works too right?

```suggestion
  if (!currentAccount?.id) && itemLoginSchemaType) {
```